cmd/create/cluster/cmd.go

```diff
@@ -404,15 +412,41 @@ func init() {
 	)
 	flags.MarkHidden("properties")
 
+	flags.StringVar(
+		&args.mode,
+		"mode",
+		"",
+		"How to perform the operation. Valid options are:\n"+
+			"auto: Roles will be created using the current AWS account\n"+
+			"manual: Role files will be saved in the current directory",
+	)
+	Cmd.RegisterFlagCompletionFunc("mode", modeCompletion)
+
+	flags.StringVar(
+		&args.operatorRolesPermissionsBoundary,
+		"operator-roles-permissions-boundary",
+		"",
+		"The ARN of the policy that is used to set the permissions boundary for the operator roles.",
+	)
+
 	interactive.AddFlag(flags)
 	output.AddFlag(Cmd)
 }
 
+func modeCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return modes, cobra.ShellCompDirectiveDefault
+}
+
 func run(cmd *cobra.Command, _ []string) {
 	reporter := rprtr.CreateReporterOrExit()
 	logger := logging.CreateLoggerOrExit(reporter)
 	var err error
 
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && !cmd.Flags().Changed("mode") {
+		interactive.Enable()
+	}
```

Comment:
I would remove this. If the user doesn't enter `--mode` we should just output the necessary `rosa` commands at the end and leave it at that. We shouldn't force the `mode` here.